### PR TITLE
feat(sdk): upgrade Ecotone SDK

### PR DESCRIPTION
**Pull request by 0x6e616d**
Following the Ecotone upgrade, I update our SDK source code.

Ref: https://github.com/ethereum-optimism/optimism/tree/develop/packages/sdk

I've tested some functionalities: deposit/withdraw eth/nativetoken/erc20

### CODE REVIEW

File: packages/tokamak/sdk/package.json [hunk 1]
```diff
@@ -67,7 +67,8 @@
     "@tokamak-network/core-utils": "workspace:*",
     "lodash": "^4.17.21",
     "merkletreejs": "^0.3.10",
-    "rlp": "^2.2.7"
+    "rlp": "^2.2.7",
+    "semver": "^7.6.0"
```
Comment (boohyung) [inline packages/tokamak/sdk/package.json:71]: Where do we use it?

Comment (0x6e616d) [reply packages/tokamak/sdk/package.json:71]: OP team used [here](https://github.com/tokamak-network/tokamak-thanos/pull/181/files#diff-21be87624c7a8a8ca2cee114f74f8846c0b99fc2e35466f0451175659111010fR266)

File: packages/tokamak/sdk/src/cross-chain-messenger.ts [hunk 2]
```diff
@@ -239,6 +247,29 @@ export class CrossChainMessenger {
...
+    ) {
+      // Only really relevant for certain SDK tests where the portal is not deployed. We should
+      // probably just update the tests so the portal gets deployed but feels like it's out of
+      // scope for the FPAC changes.
+      return false
+    } else {
+      return semver.gte(
+        await this.contracts.l1.OptimismPortal.version(),
+        '3.0.0'
+      )
```
Comment (0x6e616d) [inline packages/tokamak/sdk/src/cross-chain-messenger.ts:269]: @boohyung `semver` is used here

Comment (boohyung) [reply packages/tokamak/sdk/src/cross-chain-messenger.ts:269]: Got it! Thanks for your kindness!

**Review by boohyung — APPROVED** [review]
+1

**Pull request merged**